homeassistant/components/bosch_alarm/config_flow.py

```diff
@@ -0,0 +1,160 @@
+    """
+    panel = Panel(
+        host=data[CONF_HOST],
+        port=data[CONF_PORT],
+        automation_code=data.get(CONF_PASSWORD),
+        installer_or_user_code=data.get(CONF_INSTALLER_CODE, data.get(CONF_USER_CODE)),
+    )
+
+    try:
+        await panel.connect(load_selector)
+    finally:
+        await panel.disconnect()
+
+    return (panel.model, panel.serial_number)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Bosch Alarm."""
+
+    VERSION = 1
+    entry: config_entries.ConfigEntry | None = None
+    data: dict[str, Any] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
+            try:
+                # Use load_selector = 0 to fetch the panel model without authentication.
+                (model, _) = await try_connect(user_input, 0)
+            except (
+                OSError,
+                ConnectionRefusedError,
+                ssl.SSLError,
+                asyncio.exceptions.TimeoutError,
+            ):
+                _LOGGER.exception("Connection Error")
+                errors["base"] = "cannot_connect"
+            except Exception:
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                self.data = user_input
+                self.data[CONF_MODEL] = model
+                return await self.async_step_auth()
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self.add_suggested_values_to_schema(
+                STEP_USER_DATA_SCHEMA, user_input
+            ),
+            errors=errors,
+        )
+
+    async def async_step_auth(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the auth step."""
+        errors = {}
+
+        # Each model variant requires a different authentication flow
+        if "Solution" in self.data[CONF_MODEL]:
+            schema = STEP_AUTH_DATA_SCHEMA_SOLUTION
+        elif "AMAX" in self.data[CONF_MODEL]:
+            schema = STEP_AUTH_DATA_SCHEMA_AMAX
+        else:
+            schema = STEP_AUTH_DATA_SCHEMA_BG
+
+        if user_input is not None:
+            self.data.update(user_input)
+            try:
+                (model, serial_number) = await try_connect(
+                    self.data, Panel.LOAD_EXTENDED_INFO
+                )
+            except (PermissionError, ValueError):
+                errors["base"] = "invalid_auth"
+                _LOGGER.exception("Authentication Error")
+            except (
+                OSError,
+                ConnectionRefusedError,
+                ssl.SSLError,
+                asyncio.exceptions.TimeoutError,
```

Comment:
```suggestion
                TimeoutError,
```
that should be available in the standard space